### PR TITLE
Add Jordan holidays

### PR DESCRIPTION
**Pull request by yousefdev20**
# Contributing a new country ?

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/spatie/holidays/pulls) for the same country?

1. Create a new class in the Countries directory. It should extend the Country class.
2. Add a test for the new country in the tests directory.
3. Run the tests so a snapshot gets created.
4. Verify the result in the newly created snapshot is correct.

### CODE REVIEW

File: src/Calendars/Islamic/arafat-holidays.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+return [
+
+    'jo' => [
```
Comment (Nielsvanpach) [inline src/Calendars/Islamic/arafat-holidays.php:5]: Can you use class consts for this? Have a look at Egypt or Bahrain

File: src/Concerns/IslamicHolidays.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Spatie\Holidays\Concerns;
+
+use Carbon\CarbonImmutable;
+
+trait IslamicHolidays
```
Comment (Nielsvanpach) [inline src/Concerns/IslamicHolidays.php:7]: We already have an IslamicCalendar trait which you can use

File: src/Concerns/WeekendHolidays.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Spatie\Holidays\Concerns;
+
+use RuntimeException;
+use Carbon\CarbonPeriod;
+use Carbon\CarbonImmutable;
+
+trait WeekendHolidays
```
Comment (Nielsvanpach) [inline src/Concerns/WeekendHolidays.php:9]: I think the Observable trait is what you need

**Review by Nielsvanpach — CHANGES_REQUESTED** [review]

File: src/Calendars/Islamic/arafat-holidays.php (hunk 1 above)
Comment (yousefdev20) [reply src/Calendars/Islamic/arafat-holidays.php:5]: @Nielsvanpach 
Why I used this approach:

1. It's hard to read around 200 lines of code to read the elements.
2. in some countries the holidays are shared or in the same region.
3. at the end the Islamic directory is loadable.

File: src/Concerns/IslamicHolidays.php (hunk 2 above)
Comment (yousefdev20) [reply src/Concerns/IslamicHolidays.php:7]: This trits depends on the constants variable and I explained the issue in the above replay.

File: src/Calendars/Islamic/arafat-holidays.php (hunk 1 above)
Comment (Nielsvanpach) [reply src/Calendars/Islamic/arafat-holidays.php:5]: 2 and 3 seem valid points.

Although, using `jo` as a key doesn't feel like the way to go for being shareable across countries?

Also, if you have a look at Turkey, I prefer that format. What the benefit of using the extra `date` key? 

Also, for multiday holidays it's possible to have it twice in a single year:

```php
1999 => '01-19',
2000 => [
    '01-08',
     '12-27',
],
```

Comment (yousefdev20) [reply src/Calendars/Islamic/arafat-holidays.php:5]: `jo` it may change,

`date` is more readable,

The third note is valid.

**Pull request closed without merging**